src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -18,105 +19,78 @@
-            options.OnRecoveryError += db.HandleOnRecoveryError;
-
-            options.SchemaVersion = 1;
-            options.TransactionsMode = TransactionsMode.Lazy;
-            options.ForceUsing32BitsPager = db.Configuration.Storage.ForceUsing32BitsPager;
-            options.TimeToSyncAfterFlashInSeconds = db.Configuration.Storage.TimeToSyncAfterFlashInSeconds;
-            options.NumOfCocurrentSyncsPerPhysDrive = db.Configuration.Storage.NumOfCocurrentSyncsPerPhysDrive;
-
-
-            _environment = new StorageEnvironment(options);
-            var databaseName = db.Name;
-
-            _logger = LoggingSource.Instance.GetLogger<SubscriptionStorage>(databaseName);
-            _subscriptionsSchema.DefineKey(new TableSchema.SchemaIndexDef
-            {
-                StartIndex = 0,
-                Count = 1
-            });
+            _serverStore = serverStore;
+            _logger = LoggingSource.Instance.GetLogger<SubscriptionStorage>(db.Name);
         }
 
         public void Dispose()
         {
             foreach (var state in _subscriptionStates.Values)
             {
                 state.Dispose();
-            }
-            _environment.Dispose();
+            }            
         }
 
         public void Initialize()
         {
-            var transactionPersistentContext = new TransactionPersistentContext();
-            using (var tx = _environment.WriteTransaction(transactionPersistentContext))
-            {
-                tx.CreateTree(SubscriptionSchema.IdsTree);
-                _subscriptionsSchema.Create(tx, SubscriptionSchema.SubsTree, 16);
-
-                tx.Commit();
-            }
+          
         }
 
-        public unsafe long CreateSubscription(BlittableJsonReaderObject criteria, long ackEtag = 0)
+        // todo: maybe switch etag in subscriptions to ChangeVector...
+        public async Task<long> CreateSubscription(BlittableJsonReaderObject criteria, long ackEtag = 0)
         {
             // Validate that this can be properly parsed into a criteria object
             // and doing that without holding the tx lock
-            JsonDeserializationServer.SubscriptionCriteria(criteria);
-
-            var transactionPersistentContext = new TransactionPersistentContext();
-            using (var tx = _environment.WriteTransaction(transactionPersistentContext))
+            var criteriaInstance = JsonDeserializationServer.SubscriptionCriteria(criteria);
+            
+            var command = new CreateSubscriptionCommand(_db.Name)
             {
-                var table = tx.OpenTable(_subscriptionsSchema, SubscriptionSchema.SubsTree);
-                var subscriptionsTree = tx.ReadTree(SubscriptionSchema.IdsTree);
-                var id = subscriptionsTree.Increment(SubscriptionSchema.Id, 1);
+                Criteria = criteriaInstance,
+                InitialChangeVector = GetChangeVectorFromEtag(ackEtag)
```

Comment:
You can't assume that you'll run it on the same server as the one that has you go the etag from, so we are better off with change vector in the params in the first place.